### PR TITLE
90 add search

### PR DESCRIPTION
**Pull request by yringler**
Finish up the search thing

### CODE REVIEW

File: app/lib/tabs/search-tab.dart [hunk 1]
```diff
@@ -0,0 +1,127 @@
+import 'package:bloc_pattern/bloc_pattern.dart';
+import 'package:flutter/material.dart';
+import 'package:inside_chassidus/routes/player-route/index.dart';
+import 'package:inside_chassidus/routes/search-section-route.dart';
+import 'package:inside_chassidus/tabs/widgets/simple-media-list-widgets.dart';
+import 'package:inside_data/inside_data.dart';
+
+//TODO: add necessary navigator stuff; see MediaListTabNavigator
```
Comment (efraimbart) [inline app/lib/tabs/search-tab.dart:8]: Without this the Android (and perhaps iOS?) system back button will misbehave

Comment (yringler) [reply app/lib/tabs/search-tab.dart:8]: It's a question of priority. If back button closes the app, or moves up the library hierchy - it's not ideal, but after once or twice people will remember how it works.
As long as it's kinda justifiable, I'll send an APK to Shaul and see if he wants to prioritize this over getting started on home page / other PR etc

Comment (yringler) [reply app/lib/tabs/search-tab.dart:8]: Ok, now I see what you mean :sweat_smile: 
Yeah, I'll take care of this.

Comment (yringler) [reply app/lib/tabs/search-tab.dart]: @efraimbart  I added "back goes from library to search"!
Turns out that "back button navigates up library sections" is a bug - I had code to only navigate to sections user had navigated to, but at some point that was broken.
Once that was fixed, everything fell into place.

Removed default home page from the pages stack (now, it's only there if there's no other pages to show)

Which means that - if search clicks to library, a back would close the app - so main can now catch that, and move back to the previous tab (in this case, search) (or, if user goes from home page to subsection... ok, that's a bug, I'll fix that now :sweat_smile: )

See [restore back behaviour](https://github.com/yringler/inside-app/pull/115/commits/65b61b914f0b9a05c7f85e54f1d36073e5a7f8c0)

And [home page only if empty](https://github.com/yringler/inside-app/pull/115/commits/5bae001aa25b96bc9bca81d94b15a1a1f5885079#diff-f327ddcc9237ecb0dbe1f1927d08c418422fe614a33a718ead480dda0f5233c0R44)

And [go back to a previous tab](https://github.com/yringler/inside-app/pull/115/commits/5bae001aa25b96bc9bca81d94b15a1a1f5885079#diff-43a41aed8f4bf51ecb85660250a9e01f6b783178ba5d5317606bda12467ddb82R218)

**Pull request closed without merging**